src/NuGet.Core/NuGet.Configuration/Settings/Items/PackageSourceMappingSourceItem.cs

```diff
@@ -157,7 +157,7 @@ internal override void Update(SettingItem other)
 
             base.Update(other);
 
-            Dictionary<PackagePatternItem, PackagePatternItem> otherPatterns = packageSourceMappingSourceItem.Patterns.ToDictionary(c => c, c => c);
+            Dictionary<PackagePatternItem, PackagePatternItem> otherPatterns = packageSourceMappingSourceItem.Patterns.GroupBy(item => item).ToDictionary(c => c.Key, c => c.Key);
```

Comment:
Do we have a test? Sounds like we might, if we don't we probably need one.

This is a low level API, so it should be easily testable.

Reply:
Seems existing tests did not cover duplicates, so I added `PackageSourceMappingSourceItemTests.Update_WithDuplicatePatterns_WritesSinglePattern`

Verified it threw the exception with the previous/existing code (See screenshot).
![image](https://github.com/NuGet/NuGet.Client/assets/49205731/22d7836c-ade3-4402-9752-c5e93dbe77d8)